addons/event_sale/wizard/event_configurator.py

```diff
@@ -11,3 +12,35 @@ class EventConfigurator(models.TransientModel):
     product_id = fields.Many2one('product.product', string="Product", readonly=True)
     event_id = fields.Many2one('event.event', string="Event")
     event_ticket_id = fields.Many2one('event.event.ticket', string="Event Ticket")
+    has_available_events = fields.Boolean(
+        "Has Available Events",
+        compute="_compute_has_available_events",
+        help="Technical field used to know if the user has at least one event that can be selected and used to configure event tickets on sale order lines")
+
+    @api.model
+    def default_get(self, fields):
```

Comment:
Why not trying to make ``event_id`` and ``event_ticket_id`` computed editable fields ? That way instead of doing some logic in default_get we could do it at field level (if change product -> could update event and ticket if not set or nor matching product anymore). What do you think ?